crates/turborepo-filewatch/src/package_watcher.rs

```diff
@@ -443,6 +516,47 @@ impl Subscriber {
     }
 }
 
+async fn discovery_packages(repo_root: AbsoluteSystemPathBuf) -> PackageState {
+    // If we're rediscovering everything, we need to rediscover the package manager.
+    // It may have changed if a lockfile changed or package.json changed.
+    let discovery = match LocalPackageDiscoveryBuilder::new(repo_root.clone(), None, None).build() {
+        Ok(discovery) => discovery,
+        Err(e) => return PackageState::NoPackageManager(e.to_string()),
+    };
+    let initial_discovery = match discovery.discover_packages().await {
+        Ok(discovery) => discovery,
+        // If we failed the discovery, that's fine, we've reset the values, leave them as None
```

Comment:
Lets just get rid of the references to None here since they aren't relevant now that we've pulled out the fn

Reply:
Yup, done.